web/gui/main.js

```diff
@@ -2819,89 +2819,43 @@ function versionLog(msg) {
     document.getElementById('versionCheckLog').innerHTML = msg;
 }
 
-function getNetdataCommitIdFromVersion() {
-    var s = options.version.split('-');
+// New way of checking for updates, based only on versions
 
-    if (s.length !== 3) {
-        return null;
-    }
-    if (s[2][0] === 'g') {
-        var v = s[2].split('_')[0].substring(1, 8);
-        if (v.length === 7) {
-            versionLog('Installed git commit id of netdata is ' + v);
-            document.getElementById('netdataCommitId').innerHTML = v;
-            return v;
-        }
+function versionsMatch(v1, v2) {
+    if (v1 == v2) {
+        return true;
+    } else {
+        var s1=v1.split('-');
+        var s2=v2.split('-');
+        if (s1.length !== s2.length) return false;
+        if (s1.length === 4) s1.pop();
+        if (s2.length === 4) s2.pop();
+        return (s1.join('-') === s2.join('-'));
     }
-    return null;
-}
-
-function getNetdataCommitId(force, callback) {
-    versionLog('Downloading installed git commit id from netdata...');
-
-    $.ajax({
-        url: 'version.txt',
-        async: true,
-        cache: false,
-        xhrFields: {withCredentials: true} // required for the cookie
-    })
-        .done(function (data) {
-            data = data.replace(/(\r\n|\n|\r| |\t)/gm, "");
-
-            var c = getNetdataCommitIdFromVersion();
-            if (c !== null && data.length === 40 && data.substring(0, 7) !== c) {
-                versionLog('Installed files commit id and internal netdata git commit id do not match');
-                data = c;
-            }
-
-            if (data.length >= 7) {
-                versionLog('Installed git commit id of netdata is ' + data);
-                document.getElementById('netdataCommitId').innerHTML = data.substring(0, 7);
-                callback(data);
-            }
-        })
-        .fail(function () {
-            versionLog('Failed to download installed git commit id from netdata!');
-
-            if (force === true) {
-                var c = getNetdataCommitIdFromVersion();
-                if (c === null) {
-                    versionLog('Cannot find the git commit id of netdata.');
-                }
-                callback(c);
-            } else {
-                callback(null);
-            }
-        });
 }
 
-function getGithubLatestCommit(callback) {
-    versionLog('Downloading latest git commit id info from github...');
+function getGithubLatestVersion(callback) {
+    versionLog('Downloading latest version id from github...');
 
     $.ajax({
-        url: 'https://api.github.com/repos/netdata/netdata/commits',
+        url: 'https://api.github.com/repositories/10744183/contents/packaging/version?ref=master',
```

Comment:
Why not just download this file? It's raw version is located at https://raw.githubusercontent.com/netdata/netdata/master/packaging/version and it is much clearer what is happening than using some magic link. Additionally you cannot run into GitHub rate limiter since it is not interacting with API.

Reply:
ok